flixster codereview1

amotor95 opened the pull request:
### Summary
MVP Features:

Pulls movies from TMDB (The Movie Database)
Displays movies in grid view
Load More button at bottom
Search Function (Search Bar, Search Button, Clear Search Button)
Sorting (Title (A-Z), Release date (Chronological), Vote average (Descending))
Movie Modal w/ Backdrop poster, Name, Description, Runtime, Release date, Genres
Movie Cards Resize with Window
Stretch Features:

Embedded movie trailer player in modal
Can swap between available trailers
Can favorite movies & Favorites page
Can mark movies as watched & Watched page
Sidebar to select between now-playing, favorites, and watched page

### Tests
Loads now-playing movies on page load ✅
Search function functionality ✅
All three sorting modes functionality ✅
Load more button functionality (tested with now-playing/home, search, while sorted, and search & while sorted) ✅
Clear button functionality (appears after a search, disappears after clear, resets the search bar and sorting mode, shows now-playing) ✅
Movie Modal functionality (tested with movies w/ no trailer or backdrop photo, alt text works) ✅
Favorites page functionality (tested liking from now-playing, searched movies, favorites, and watched, also tested while sorted) ✅
Watchlist page functionality (tested marking as watched from now-playing, search movies, favorites, and watched, also tested while sorted) ✅
Sidebar functionality (tested swapping between pages after searching, sorting, unliking/liking, un-watching/watching) ✅

Commit:
Finished searching and sorting

Commit:
Finished modal, add modal content

Commit:
Added trailers viewer

Commit:
first deploy

Commit:
Trying to fix movie image for live deploy

Commit:
remade build

Commit:
Moved movie.png for relative path

Commit:
None ordering returns to original ordering

Commit:
Finished favorites and watched

Commit:
review1-branch-push

Commit:
Refactor movie fetching and processing functions
- Rename fetchMovieList() to fetchAndProcessMoviesByIDList() for clarity
- Split fetchAndProcessMoviesByIDList() into fetchMovieByID() and processMoviesByID()
- Rename fetchSearchMovies() to fetchAndProcessMoviesBySearch() for clarity
- Split fetchAndProcessMoviesBySearch into buildMovieSearchURL(), fetchMoviesBySearch(), and processMoviesBySearch()
- Refactor sortMovies() to sortMovieOrder() for clarity and replace if/else with switch/case for readability
- Update useEffect dependencies [mode, pageCleared] to use switch/case for readability
These changes improve code readability and maintainability by clarifying function purposes and enhancing control flow readability.

Commit:
Fix bug where processMoviesBySearch() was duplicating previous movies in Load More

amotor95 committed:
Merge pull request #2 from amotor95/jackmccl/flixster/styling-and-polishing

Refactoring Movie Fetch Functions and Sorting

Commit:
fix - fixed search bar not clearing when swapping from now playing to favorites/watched and then back with text in searchText

Commit:
Fix - modal opening in movie card due to moviecard:hover {scale: 105%}, moved modal component to movielist component

anykac commented on src/MovieBox.jsx:
```diff
@@ -0,0 +1,271 @@
+        // Split movies dict into keys (id) and values (movie)
+        let movieEntries = Object.entries(movies)
+        // Listened to feedback about using switch/case for repeated if/else if or where it's easier to read
+        switch (sortMode) {
```
yay switch case :D

anykac commented on src/MovieBox.jsx:
```diff
@@ -0,0 +1,271 @@
+                    const right = right_entry[1]
+                    left.date = new Date(left.release_date)
+                    right.date = new Date(right.release_date)
+                    return right.date - left.date
```
I think readability wise this can be condensed down
```suggestion
                movieEntries.sort((left_entry, right_entry) => {
                    const left_date = new Date( left_entry[1].release_date)
                    const right_date = new Date(right_entry[1].release_date)
                    return right_date - left_date
```

anykac commented on src/MovieBox.jsx:
```diff
@@ -0,0 +1,271 @@
+                    const left = left_entry[1]
+                    const right = right_entry[1]
+                    return right.vote_average-left.vote_average
+                })
```
Since the left/right consts are only used once after and the left_entry and right_entry are well named, for conciseness this could also be condensed to:
```suggestion
                movieEntries.sort((left_entry, right_entry) => {
                    return right_entry[1].vote_average-left_entry[1].vote_average
                })
```

anykac commented on src/MovieBox.jsx:
```diff
@@ -0,0 +1,271 @@
+//review1-branch
+
+const MovieBox = (mode) => {
+    // Movies is a dict
+    // Decoupled movies and ids to allow O(1) movie retrieval
```
😎 🌟

anykac commented on src/MovieBox.jsx:
```diff
@@ -0,0 +1,271 @@
+        let newFavorites = [...favorites]
+        if (newFavorites.includes(movieID)) {
+            newFavorites = newFavorites.filter( (id) => {return id !== movieID})
+            console.log(movieID)
```
totally fine and not an expectation for these assignments for codepath, but for future dev practice, we typically remove testing logs before landing code changes.

anykac commented on src/MovieBox.jsx:
```diff
@@ -0,0 +1,271 @@
+                    break
+            }
+        }
+        const promises = movieIDList.map((movieID) => fetchMovieByID(movieID))
```
nit: promises is a pretty vague name alone, it could potentially be renamed to something like moviePromises to indicate the info it has.

Commit:
Search input, search button, clear button, sort dropdown - restyled

Commit:
fix-footer at bottom of content

anykac reviewed (commented):
Great work getting this much done early! Thanks for implementing the changes suggested from last week's assignment. Looking forward to seeing your finalized submission product at the EOW :)

Commit:
CSS variable styling - colors controlled by variables in App.css

Commit:
Fix - load more was showing even when showing last page of results

Commit:
Fix - condensed sort functions for readability
fix - removed testing console logs
fix - renamed promises to moviePromises for readability

amotor95 committed:
Merge pull request #3 from amotor95/jackmccl/flixster/styling-and-polishing

Jackmccl/flixster/styling and polishing
